src/ai/ai.cpp

```diff
@@ -1692,27 +1710,26 @@ dcon::cb_type_id pick_fabrication_type(sys::state& state, dcon::nation_id from,
 	}
 }
 
-bool valid_construction_target(sys::state& state, dcon::nation_id from, dcon::nation_id target) {
+static inline bool valid_construction_target(sys::state& state, dcon::nation_id from, dcon::nation_id target) {
+	// Copied from commands.cpp:can_fabricate_cb()
 	if(from == target)
 		return false;
-
+	if(state.world.nation_get_constructing_cb_type(from))
+		return false;
 	auto ol = state.world.nation_get_overlord_as_subject(from);
 	if(state.world.overlord_get_ruler(ol) && state.world.overlord_get_ruler(ol) != target)
 		return false;
-	auto sl = state.world.nation_get_in_sphere_of(target);
-	if(sl == from)
-		return false;
-	if(nations::are_allied(state, target, from))
-		return false;
-	if(nations::are_allied(state, sl, from))
+	if(state.world.nation_get_in_sphere_of(target) == from)
 		return false;
-	if(state.world.nation_get_military_score(target) > state.world.nation_get_military_score(from))
+	if(military::are_at_war(state, target, from))
 		return false;
-	if(state.world.nation_get_owned_province_count(target) <= 3)
+	
+	if(estimate_defensive_strength(state, target) * 0.5f > estimate_defensive_strength(state, from))
 		return false;
-	if(military::are_at_war(state, target, from))
+	// Allows small one-province-minors to declare war on other states, and bigger countries to ignore small one-province-minors
```

Comment:
You have missed the point of the line you changed. That line didn't prevent small nations from fabricating -- it prevented nations from wasting infamy by fabricating on them

Reply:
it allows small nations to fight each other

Reply:
Nations wasting full infamy amounts for 1 or 2 provinces should be avoided. The way things were let the ai of small nations fabricate on people, but prevented nations for wasting buckets of infamy on a tiny number of provinces. Note that they can still end up fighting if they have cbs from other sources; they just won't spend infamy

Reply:
No no no I wanted to make it so only SMALL nations fight other SMALL nations

Reply:
No thanks

Reply:
Why not????? It's the most logical thing to do, for e.x Indian non-sphered nations

Reply:
Because if a small nation has the opportunity, it should try to take a chunk off a larger neighbor.